sqlglot/dialects/spark2.py

```diff
@@ -126,6 +126,8 @@ class Spark2(Hive):
 
     class Tokenizer(Hive.Tokenizer):
         HEX_STRINGS = [("X'", "'"), ("x'", "'")]
+        # https://spark.apache.org/docs/latest/sql-ref-literals.html
+        ESCAPE_FOLLOW_CHARS = ["0", "b", "n", "r", "t", "Z", "%", "_"]
```

Comment:
This may need to go in `Hive` instead of `Spark2`, I'm observing similar behavior:

```
0: jdbc:hive2://localhost:10000> select '\g';
+------+
| _c0  |
+------+
| g    |
+------+
```